draft-ietf-oauth-selective-disclosure-jwt.md

```diff
@@ -1147,15 +1147,59 @@ Decoy digests increase the size of the SD-JWT. The number of decoy digests (or w
-
-To prevent these types of linkability, various methods, including but not limited to the following ones can be used:
-
-- Use advanced cryptographic schemes, outside the scope of this specification.
-- Issue a batch of SD-JWTs to the Holder to enable the Holder to use a unique SD-JWT per Verifier. This only helps with Verifier/Verifier unlinkability.
-
+Unlinkability is a property whereby adversaries are prevented from correlating
+credential presentations of the same user (beyond the user's consent).
+Otherwise, an adversary might be able to learn more about the user than the user
+intended to disclose, for example:
+
+ * Cooperating Verifiers might want to track users across services to build
+   advertising profiles.
+ * Issuers might want to track where users present their credentials to enable
+   surveillance.
+ * After a data breach at multiple Verifiers, publicly available information
+   might allow linking identifiable information presented to Verifier A with
+   originally anonymous information presented to Verifier B, therefore revealing
+   the identities of users of Verifier B.
+
+The following types of unlinkability are considered here:
+
+ * Verifier/Verifier Unlinkability: Two colluding Verifiers should not be able to
+   learn that they have received presentations from the same user.
+ * Issuer/Verifier Unlinkability (Honest Verifier): An Issuer of a credential
+   should not be able to know that a user presented the credential to a certain
+   Verifier that is behaving according to protocol.
+ * Issuer/Verifier Unlinkability (Colluding/Compromised Verifier): An Issuer of a
+   credential should not be able to tell that a user presented the credential to
+   a certain Verifier, even if the Verifier colludes with the Issuer or becomes
+   compromised and leaks stored credentials from presentations.
+
+In all cases, unlinkability is limited to cases where the disclosed claims do
+not contain information that directly or indirectly identifies the user. For
+example, when a tax ID is contained in the disclosed claims, the Issuer and
+Verifier can easily link the user's transactions. However, when the user only
+discloses a birthdate to one Verifier and a postal code to another Verifier, the two Verifiers should not be able to determine that they were interacting with the same user.
+
+Issuer/Verifier unlinkability with a colluding or compromised Verifier cannot be
+achieved in salted-hash based approaches to credentials, such as SD-JWT, as the
+issued credential is directly forwarded to the Verifier, who can forward it to
+the malicious Issuer.
+
+Issuer/Verifier unlinkability with an honest Verifier can be achieved. A
+'phone-home' from the Verifier to the Issuer, for example, for a revocation
+check, SHOULD be avoided, or, if that is not possible, MUST be done in a
+privacy-preserving way where details about the credential are not revealed to
+the Issuer. (Even then, timing of such requests might remain as a side-channel.)
```

Comment:
How about this?
```suggestion
Contrary to that, Issuer/Verifier unlinkability with an honest Verifier can generally be achieved.
However, a callback from the Verifier to the Issuer, for example, for a revocation
check, can reveal information about the use of the credential to the Verifier. Therefore,
such callbacks SHOULD be avoided, or, if that is not possible, MUST be done in a
privacy-preserving way where details about the credential are not revealed to
the Issuer. (Even then, timing of such requests might remain as a side-channel.)
```

Reply:
@peppelinux

Reply:
looks good, and I would also contribute with the followings

`However, a callback from the Verifier to the Issuer, for example, for a revocation
check, can reveal information about the use of the credential to the Verifier.` -> `A callback from the Verifier to the Issuer, such as a revocation check, could potentially disclose information about the credential's usage to the Verifier.`

and also

`Therefore,
such callbacks SHOULD be avoided, or, if that is not possible, MUST be done in a
privacy-preserving way where details about the credential are not revealed to
the Issuer. (Even then, timing of such requests might remain as a side-channel.)` -> `Therefore, such callbacks SHOULD be avoided. If they are unavoidable, they MUST be executed in a manner that preserves privacy and does not disclose details about the credential to the Issuer. However, it's important to note that the timing of such requests could potentially serve as a side-channel.`

Reply:
```suggestion
Contrary to that, Issuer/Verifier unlinkability with an honest Verifier can generally be achieved.
However, a callback from the Verifier to the Issuer, such as a revocation check, could potentially
disclose information about the credential's usage to the Verifier. Therefore, such callbacks
SHOULD be avoided. Where they are unavoidable, they MUST be executed in a manner that
preserves privacy and does not disclose details about the credential to the Issuer. It is
important to note that the timing of such requests could potentially serve as a side-channel.
```